### PR TITLE
Docker with CI integration ready

### PR DESCRIPTION
**Pull request by rayandrew**
@keychera test and background start already __implemented__

Kindly check and review, thank you
cc : @deryrahman @keychera

### CODE REVIEW

File: application.properties.sample [hunk 1]
```diff
@@ -6,6 +6,20 @@ spring.datasource.driver-class-oracle.jdbc.driver.OracleDriver
 
 # HikariCP settings
 # spring.datasource.hikari.*
-
 spring.datasource.hikari.connection-timeout=60000
```
Comment (deryrahman) [inline application.properties.sample:9]: I've alread re-add hikaricp configuration. we need this for connection pooling

**Review by deryrahman — APPROVED** [review]
just wait @keychera for travis integration 👌

File: test [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+docker-compose up -d &&
+docker-compose exec app sh ./gradlew test
```
Comment (keychera) [inline test:4]: correct me, Travis will have to call
./init
./start background
./test
for testing, is that it?
@rayandrews

**Review by keychera — APPROVED** [review]
approved, gonna add one commit to add the .travis.yml first for it to pass the check

File: test (hunk 2 above)
Comment (rayandrew) [reply test:4]: yep, just test it on your PC first before pushing in case misconfiguration @keychera

**Comment by keychera** [comment]
that is not one commit lol but okay

**Comment by rayandrew** [comment]
gotcha, all tests in CI have been **passed**. gotta merge to master branch.

cc : @deryrahman @keychera 👍 

ready to merge

**Comment by keychera** [comment]
why does the test now fail when it initially succeeded just before??

**Comment by rayandrew** [comment]
it's just a matter of permission that i think changed "unpredictable". 

all my tests are already passed somehow

**Comment by keychera** [comment]
but this is still weird
the modify travis commit
```
:test
2018-03-03 06:28:15.477  INFO 267 --- [       Thread-7] ationConfigEmbeddedWebApplicationContext : Closing org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext@46dce0f3: startup date [Sat Mar 03 06:28:06 GMT 2018]; root of context hierarchy
BUILD SUCCESSFUL in 1m 0s
3 actionable tasks: 3 executed
```
after you closed the pull request and reopened it
```
:test
2018-03-03 06:38:29.293  INFO 247 --- [       Thread-7] ationConfigEmbeddedWebApplicationContext : Closing org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext@712184b6: startup date [Sat Mar 03 06:38:20 GMT 2018]; root of context hierarchy
:test FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':test'.
> A build operation failed.
      Could not write XML test results for id.ac.itb.logistik.ditlog.DitlogApplicationTests to file /opt/ditlog-webservice/build/test-results/test/TEST-id.ac.itb.logistik.ditlog.DitlogApplicationTests.xml.
```

**Comment by rayandrew** [comment]
```
Could not write XML test results for id.ac.itb.logistik.ditlog.DitlogApplicationTests to file /opt/ditlog-webservice/build/test-results/test/TEST-id.ac.itb.logistik.ditlog.DitlogApplicationTests.xml.
```

We will see if the issue comes up again, and i will check for the fixes. 
I think the permission is changed in this case.
But i haven't see any issues again after that.

**Pull request merged**